worlds/hk/Options.py

```diff
@@ -294,15 +294,33 @@ def get_costs(self, random_source: Random) -> typing.List[int]:
             return charms
 
 
+class CharmCost(Range):
+    range_end = 6
+
+
 class PlandoCharmCosts(OptionDict):
     """Allows setting a Charm's Notch costs directly, mapping {name: cost}.
     This is set after any random Charm Notch costs, if applicable."""
     display_name = "Charm Notch Cost Plando"
     valid_keys = frozenset(charm_names)
     schema = Schema({
-        Optional(name): And(int, lambda n: 6 >= n >= 0, error="Charm costs must be integers in the range 0-6.") for name in charm_names
+        Optional(name): Or(
+            And(str, lambda n: n.startswith("random")),
+            And(int, lambda n: 6 >= n >= 0),
+            error="Charm costs must be integers in the range 0-6 or use a random prefixed string."
+            )
+        for name in charm_names
         })
 
+    def __init__(self, value):
+        self.value = {}
+        for key, data in value.items():
+            try:
+                self.value[key] = CharmCost.from_any(data).value
```

Comment:
Perhaps of note, this also means that `{Wayward Compass: high, Fragile Strength: Low, Carefree Melody: default}` is valid, which maps these to 6, 0, and 0 respectively

Reply:
I noticed and just left as an undocumented feature, though the default=0 did give me a bit of pause because people might think you can use that to set vanilla charm costs manually

Reply:
FWIW I've never heard of a single person using these

Reply:
default=0 does make me a bit nervous as well. I do think it would be good to have a way to plando a cost as vanilla without needing to know all the costs. Could be `default` or something we add.

Reply:
probably better to make a "vanilla" override that looks up the number outside of CharmCost 
and then just manually crash on "default" to limit confusion

Reply:
Yeah that seems reasonable